contrib/pkg/sets/sets.go

```diff
@@ -101,10 +122,26 @@ func (s resourceSet) Difference(set ResourceSet) ResourceSet {
 
 func (s resourceSet) Intersection(set ResourceSet) ResourceSet {
 	newSet := s.set.Intersection(set.Keys())
-	var newResources []metav1.Object
+	var newResources []ezkube.ResourceId
 	for key, _ := range newSet {
 		val, _ := s.mapping[key]
 		newResources = append(newResources, val)
 	}
 	return NewResourceSet(newResources...)
 }
+
+func (s resourceSet) Find(
```

Comment:
This feels strange to me, why are we passing a `resourceType` and an `Id` here of the same type?